### ISSUE TITLE
`blobstore.enable_signed_urls` doesn't work without `director.enable_nats_delivered_templates`

### ISSUE THREAD

**Issue by videlov** [cloudfoundry/bosh#2288]
When enabling `blobstore.enable_signed_urls` BOSH director sends signed-urls for the packages during VM update phase, but do not include the signed urls for the rendered templates. For instance, this NATS apply processed by the VM agent:

```
2020-11-09_15:31:55.28854 {"protocol":3,"method":"apply","arguments":[{"deployment":"bosh-sf","job":{"name":"bosh","templates":[{"name":"nats","version":"4d19655090e879ab051dbe10ad350d9d736bae7762e1bd0b31333d8a5f9abbdd","sha1":"sha256:035884b6dce8bdb70e6cd37de90cf9096e67e053338eb5906d48a45e3410bc9d","blobstore_id":"07a6c18a-d919-4b4c-a729-490d9e722ea9","logs":[]},{"name":"blobstore","version":"2ca8095778c1ba9e20d359db6af31c372b74b07f67f1010b2c095261742c5509","sha1":"sha256:74818955bdf8aec44cf097d33458c910e0fc8c835acad7bdc76acaeeecf0b13f","blobstore_id":"e33cf30f-32c3-469d-9dc5-92adc2234c93","logs":[]},...,"template":"nats","version":"4d19655090e879ab051dbe10ad350d9d736bae7762e1bd0b31333d8a5f9abbdd"},"index":0,"name":"bosh","id":"0b9c4b40-ee6a-48a4-82b9-c84153c8a274","az":"z1","networks":{"bosh-sf":{"type":"manual","ip":"192.168.1.21","netmask":"255.255.255.0","cloud_properties":{"application_security_groups":["asg-v2-bosh"],"subnet_name":"subnet-bosh-azure-dev20","virtual_network_name":"vnet-azure-dev20"},"default":["dns","gateway"],"dns":["168.63.129.16"],"gateway":"192.168.1.1"}},"packages":{"gonats":{"name":"gonats","version":"f58980bd4b0436ff65f588627116dfff63f346f4d13175b7ba47380ab89e08a6.1","sha1":"49dbee55ee699752f28e6755672387639f4aeeb1","blobstore_id":"8e6835a0-b9f4-47ee-6506-ffdf8f4a8b17","signed_url":"https://192.168.1.11:25250/signed/11/8e6835a0-b9f4-47ee-6506-ffdf8f4a8b17?e=86400&st=XQvbR9CCx2nObZnwKtVwokeBi_4IooL6i-Ga47UV_P4&ts=1604935915"},"director":{"name":"director","version":"f32385256198535b797059dd4990fcb3b65c0c07337990163c24275a7a29b7e1.1","sha1":"5c71dae9f32b72f874d21d491ef91c3f4fc9e612","blobstore_id":"1edf00c5-42cc-4fc9-514b-a759cb0cedfa","signed_url":"https://192.168.1.11:25250/signed/b6/1edf00c5-42cc-4fc9-514b-a759cb0cedfa?e=86400&st=syYoYm6vK4H9Lm52XPB8ziitBwk06cQrBWQnA8kChqk&ts=1604935915"},...,"dns_domain_name":"bosh","persistent_disk":120000,"template_hashes":{"nats":"c5f465570e867c78b2e3612f94678b65cd981a52","blobstore":"f541ac10f881876de3d302af210f940d94323fc4","director":"f5ae889a9674c1fbbb4f584bb40e7d8dc0301b50",...,"harden_sshd":"6bff8fd0847f38a96da095dbfbef1bb178415074"},"rendered_templates_archive":{"blobstore_id":"f85a3fbd-f7a3-4e7b-b94a-22a7d99e2cc5","sha1":"ed67b869ee5664eecb906c056c0794fc1b4a9911"},"configuration_hash":"011e08deae6b069296991d427e72c3e14843e348"}],"reply_to":"director.b9fb8db2-28c0-410a-8e01-e69e96a51a7e.61e2ba88-627f-481f-beb7-f083422455e2.ae34b22a-b6ae-4128-bde6-4f55a0dfa4bc"}
```

The problem comes when the secrets were stripped from the VM client blobstore options when `blobstore.enable_signed_urls` is enabled on the BOSH VM, in file:
`/var/vcap/bosh/etc/blobstore-dav.json`

There were missing `user` and `password`, which are needed for the basic authentication when requesting blobs via https endpoint from director blobstore.

When we redeploy a VM we get 401 Unauthorized error, e.g.:
```
Task 29678 | 11:28:08 | L installing packages: bosh/0b9c4b40-ee6a-48a4-82b9-c84153c8a274 (0) (canary) (00:05:36)
                     L Error: Action Failed get_task: Task a3f43d45-0ff4-4be8-7770-a3fd28116620 result: Applying: Applying job nats: Preparing job: Getting job source from blobstore: Getting blob from inner blobstore: Getting blob from inner blobstore: Shelling out to bosh-blobstore-dav cli: Running command: 'bosh-blobstore-dav -c /var/vcap/bosh/etc/blobstore-dav.json get f85a3fbd-f7a3-4e7b-b94a-22a7d99e2cc5 /var/vcap/data/tmp/bosh-blobstore-externalBlobstore-Get122181834', stdout: 'Error running app - Getting dav blob f85a3fbd-f7a3-4e7b-b94a-22a7d99e2cc5: Request failed, response: Response{ StatusCode: 401, Status: '401 Unauthorized' }', stderr: '': exit status 1
```

In the above `apply` example above that blob object is a rendered template (which do not have `signed_url` specified).

In the usage guide (https://bosh.io/docs/director-blobstore-signed-urls/) it is not mentioned that this feature works only when  `director.enable_nats_delivered_templates` is also enabled. If this is the case, you should mention it in there. Also stripping of secrets for the blobstore client option is not specified. It is actually documented that the blobstore secrets will be left on the VMs and manual cleanup via `instance_groups[name=bosh].properties.agent.env.bosh.blobstores: []` should be initiated.

Should the auto-removal of secrets be reconsidered [in here](https://github.com/cloudfoundry/bosh/blob/c36124d0d733e6901c969a31e523b105591b0603/src/bosh-director/lib/bosh/director/deployment_plan/steps/create_vm_step.rb#L185)?

To summarise issue/questions:
1. If the old blobstore access for rendered templates won't be adopted / supported, should we consider that `blobstore.enable_signed_urls` only works with `director.enable_nats_delivered_templates` ?
2. Would you update the `enable_signed_urls` documentation?
3. What about stripping client blobstore secrets?

Thanks,
@videlov, @OliverMautschke
